docs/rfcs/no-unusable-components-line-in-the-sand.md

```diff
@@ -0,0 +1,80 @@
+# No unusable Source components
+
+This proposes a line in the sand regarding how Source components should behave when their ideal version requires technologies that a client may not offer.
+
+## Example problem
+
+A Toggle Switch is similar to a Checkbox. Unlike a Checkbox, the state it sets should apply immediately i.e. a user does not need to click a Submit Button.
+
+It needs JavaScript to achieve this. The web platform provides no obvious default widget to fall back to – it's not just a visual treatment of standard widget.
+
+If we server render it, [users without JavaScript enabled](https://kryogenix.org/code/browser/everyonehasjs.html) (or who have not yet downloaded the bundle) will be presented with an unresponsive Toggle Switch until the JavaScript arrives.
+
+## Proposal
+
+A new line in the sand for Source: 
+
+> Do not present a user with a component they cannot use.
+
+Given the example above, this could mean either:
+
+- not displaying it to users until the JavaScript loads OR
+- identifying and displaying a non-JS fallback
+
+Note that this *does not mandate fallbacks*. A team could decide that the component is an enhancement and not critical, and so they decide to just not show it to people without javascript.
+
+But rendering it in an inoperative state would cross the line in the sand. 
```

Comment:
Would we expect consumers of Source to adhere to this line in the sand for when they're composing components of their own? Or will we just uphold this standard within components that belong to the design system? I appreciate that this is difficult to control and is probably something that we do not want to do. Thought I'd pose the question anyway.

To spread awareness of this line, I'm wondering whether we should also think about how this is documented for the components where this is required (in code, storybook or elsewhere)

Reply:
no this is strictly for source